WalletWasabi/WabiSabi/Models/InputWitnessPair.cs

```diff
@@ -0,0 +1,21 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WalletWasabi.WabiSabi.Models
+{
+	public class InputWitnessPair
+	{
+		public InputWitnessPair(OutPoint input, WitScript witness)
+		{
+			Input = input;
+			Witness = witness;
+		}
+
+		public OutPoint Input { get; }
```

Comment:
Why and `OutPoint`?  Shouldn't be the index enough?

Reply:
That's ambigous. The reader will think that it's a transaction index, while it isn't. Another benefit of out pointing is even if you lose what order you submitted your inputs you'll not have to care.